Tests/MongoSwiftTests/ClientSessionTests.swift

```diff
@@ -14,7 +14,73 @@ final class ClientSessionTests: MongoSwiftTestCase {
         super.tearDown()
     }
 
-    /// Test that sessions are properly returned to the pool when ended.
+    typealias CollectionSessionOp = (name: String, body: (MongoCollection<Document>, ClientSession?) throws -> Void)
+    typealias DatabaseSessionOp = (name: String, body: (MongoDatabase, ClientSession?) throws -> Void)
+    typealias ClientSessionOp = (name: String, body: (MongoClient, ClientSession?) throws -> Void)
+    typealias SessionOp = (name: String, body: (ClientSession?) throws -> Void)
+
+    typealias InsertOneModel = MongoCollection<Document>.InsertOneModel
+
+    // list of read only operations on MongoCollection that take in a session
+    let collectionSessionReadOps: [CollectionSessionOp] = [
```

Comment:
I moved all the operations previously just defined in the args test out here so that they can easily be used in any spec test that requires all methods that take in a session to be tested. This affects some of the old tests a little bit.

Reply:
very nice refactor!